### DISCUSSION TITLE
How to prevent JSLT removing null & empty properties

### DISCUSSION THREAD

**Discussion by cvsudheer108** [schibsted/jslt#235] [General]
Hi JSLT Team,
I found an a documentation related to this, to use Object Key Filter to prevent this from happening.
https://github.com/schibsted/jslt/blob/master/docs/api.md#object-key-filter

But I am not sure where to put this code:
Expression e = new Parser(reader)
  .withObjectFilter(filter)
  .parse();

Moreover, I am using Camel to invoke the JSLT transform.
Please throw some light on implementing this.

**Comment by larsga** [comment]
The example is for when you're using the JSLT API directly. If you're working through Camel you need to see how Camel supports it. I guess it's entirely possible you need to add it to Camel yourself. My guess is it's really easy to solve that.

**Reply by cvsudheer108** [reply]
Thank you. Will explore that.

**Comment by RahulKesharwani353** [comment]
Hey @larsga 
Expression e = new Parser(reader)
.withObjectFilter(filter)
.parse();

what is `reader` here can you brief it more, I'm using Java API

```java
        Reader reader = new Reader() {
            @Override
            public int read(char[] cbuf, int off, int len) throws IOException {
                return 0;
            }

            @Override
            public void close() throws IOException {

            }
        };

        Expression jslt = new Parser(reader)
                .withObjectFilter("true")
                .compileString(convertModel.getJsltTemplate());

        JsonNode output = jslt.apply(convertModel.getJsonInput());
```

It is not working for me. Can you please help me out

**Comment by larsga** [comment]
`Reader` is `java.io.Reader`. You need a reader that actually returns the JSLT that you want to compile. If you have the expression as a string you can use `StringReader`, for example.

**Reply by RahulKesharwani353** [reply]
error: cannot find symbol
                .parse();
                ^
  symbol:   method parse()
  location: class Parser
I'm getting this error

**Reply by RahulKesharwani353** [reply]
it is solved by using `compile` instead of `parse`, if it is fine can you update the docs too.

**Reply by larsga** [reply]
Thanks for letting me know about the mistake in the documentation! Fixed now.
